src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -198,10 +198,13 @@ public Task StartAsync()
                 // Get a connection ID from the server
                 logger.EstablishingConnection(url);
                 using (var request = new HttpRequestMessage(HttpMethod.Options, url))
-                using (var response = await httpClient.SendAsync(request))
                 {
-                    response.EnsureSuccessStatusCode();
-                    return await ParseNegotiateResponse(response, logger);
+                    request.Headers.UserAgent.Add(SendUtils.DefaultUserAgentHeader);
+                    using (var response = await httpClient.SendAsync(request))
+                    {
+                        response.EnsureSuccessStatusCode();
```

Comment:
How does it fail? I think we should add a test for this - it shouldn't be too difficult...

Reply:
Does that belong in this PR, or maybe a another one?

Reply:
I think ensuring the status is a separate PR/fix - I just commented only because you added it here but I am fine either way as long as we don't forget.

Reply:
https://github.com/aspnet/SignalR/issues/768